### PR TITLE
Ensure the modifications made through the `tribe_events_views_v2_category_title` are applied. [TEC-4110]

### PR DESCRIPTION
**Pull request by EugeneKyale**
This PR ensures that users are able to use the `tribe_events_views_v2_category_title` filter to modify the page title on category archive pages.

Ticket : https://theeventscalendar.atlassian.net/browse/TEC-4110
Artifact : https://www.loom.com/share/b62845c18ef243a7bad65411342a207f

### CODE REVIEW

File: src/Tribe/Views/V2/Template/Title.php [hunk 1]
```diff
@@ -140,7 +140,7 @@ public function build_title( $current_title = '', $depth = true ) {
 		}
 
 		$term = $context->get( TEC::TAXONOMY, false );
-		if ( false !== $term && $depth ) {
+		if ( false !== $term && $depth = true ) {
```
Comment (Camwyn) [inline src/Tribe/Views/V2/Template/Title.php:143]:
```suggestion
		if ( false !== $term && $depth === true ) {
```

Comment (Camwyn) [reply src/Tribe/Views/V2/Template/Title.php:143]: You are setting it here rather than checking it!

File: src/Tribe/Views/V2/Template/Title.php [hunk 2]
```diff
@@ -140,7 +140,7 @@ public function build_title( $current_title = '', $depth = true ) {
 		}
 
 		$term = $context->get( TEC::TAXONOMY, false );
-		if ( false !== $term && $depth ) {
+		if ( false !== $term && true !== $depth ) {
```
Comment (Camwyn) [inline src/Tribe/Views/V2/Template/Title.php:143]: I don't think this is the right solution either.

Comment (EugeneKyale) [reply src/Tribe/Views/V2/Template/Title.php:143]: @Camwyn Any suggestions?

File: src/Tribe/Views/V2/Template/Title.php [hunk 3]
```diff
@@ -366,7 +366,7 @@ protected function build_day_title( $event_date ) {
 	 * @return string The built category archive title.
 	 */
 	protected function build_category_title( $title, $cat ) {
-		$new_title = '<a href="' . esc_url( tribe_get_events_link() ) . '">' . $title . '</a>';
```
Comment (bordoni) [inline src/Tribe/Views/V2/Template/Title.php:369]: I think the category title still needs the link, we might need to figure out how to exclude it in the specific places we need it to be removed.

**Review by bordoni — CHANGES_REQUESTED** [review]

File: src/Tribe/Views/V2/Template/Title.php (hunk 3 above)
Comment (EugeneKyale) [reply src/Tribe/Views/V2/Template/Title.php:369]: @bordoni I don't think there's a use case for having a linked category on the page title. I can't find anywhere else where this title is being used. 
Screenshot 2022-01-04 at 22 14 49

Comment (bordoni) [reply src/Tribe/Views/V2/Template/Title.php:369]: Ahh i see that is good to know. You are correct here, we need revise the depth problem that stephen brought up.

**Review by bordoni — APPROVED** [review]

**Review by bordoni — APPROVED** [review]

**Review by bordoni — APPROVED** [review]

**Pull request merged**